src/Education/ErrorList/SonarErrorListEventProcessor.cs

```diff
@@ -0,0 +1,66 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft;
+using Microsoft.VisualStudio.Shell.TableControl;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Infrastructure.VS;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.Education
+{
+    namespace SonarLint.VisualStudio.Education.ErrorList
+    {
+        internal class SonarErrorListEventProcessor : TableControlEventProcessorBase
```

Comment:
This class is a processor that gets called for various WPF events that occur on the error list table. The base class defines all of the events that can be handled; we just need to override the ones we are interested in.

Currently there is only one - the pre-navigate event, which is the one that VS uses to process the hyperlink click.